app/models/organisation.rb

```diff
@@ -156,16 +157,20 @@ def editable_duplicate_sales_logs_sets
     sales_logs.after_date(FormHandler.instance.sales_earliest_open_for_editing_collection_start_date).duplicate_sets.map { |array_str| array_str ? array_str.map(&:to_i) : [] }
   end
 
-  def recently_absorbed_organisations_grouped_by_merge_date
-    return unless absorbed_organisations.present? && absorbed_organisations.merged_during_open_collection_period.present?
+  def organisations_absorbed_during_displayed_collection_period_grouped_by_merge_date
+    return unless absorbed_organisations.present? && absorbed_organisations.merged_during_displayed_collection_period.present?
```

Comment:
do we think that when this method is called we will hold the absorbed_organisations in memory already? I'm guessing not since `#merged_during_displayed...` is a scope?
if so `#present?` will convert those things to ruby objects, whereas `#exists?` would generate a DB query that returns true/false directly. That might be preferable?
Or maybe if the result of the query is cached by active record anyway it makes no difference (given we then do some more queries on line 163?

Also, given `merged_during_displayed_collection_period` is where clause a scope, I imagine that `absorbed_organisations.present?` is redundant?

Reply:
Yup, these suggestions make sense!